mne/morph.py

```diff
@@ -0,0 +1,731 @@
+                 'mri_shape': self.morph_data['mri_shape_to']})
+
+        return _apply_morph_data(self, stc_from, verbose=verbose)
+
+    def __repr__(self):  # noqa: D105
+        s = None
+        if self.kind is not None:
+            s = "%s" % self.kind
+            s += ", subject_from : %s" % self.subject_from
+            s += ", subject_to : %s" % self.subject_to
+            if self.kind == 'volume':
+                s += ", niter_affine : {}".format(self.niter_affine)
+                s += ", niter_sdr : {}".format(self.niter_sdr)
+                s += ", grid_spacing : {}".format(self.grid_spacing)
+
+            elif self.kind == 'surface':
+                s += ", smooth : %s" % self.smooth
+                s += ", xhemi : %s" % self.xhemi
+
+        return "<SourceMorph  |  %s>" % s
+
+    @verbose
+    def save(self, fname, verbose=None):
+        """Save the morph for source estimates to a file.
+
+        Parameters
+        ----------
+        fname : str
+            The stem of the file name. '-morph.h5' will be added if fname does
+            not end with '.h5'
+        verbose : bool | str | int | None
+            If not None, override default verbose level (see
+            :func:`mne.verbose` and
+            :ref:`Logging documentation <tut_logging>`
+            for more). Defaults to self.verbose.
+        """
+        logger.info('saving morph...')
+        if not fname.endswith('.h5'):
+            fname = '%s-morph.h5' % fname
+
+        write_hdf5(fname, self.__dict__,
+                   overwrite=True)
+        logger.info('[done]')
+
+    def as_volume(self, stc, fname=None, mri_resolution=False):
+        """Return volume source space as Nifti1Image and/or save to disk.
+
+        Parameters
+        ----------
+        stc : VolSourceEstimate | SourceEstimate | VectorSourceEstimate
+            Data to be transformed
+        fname : str | None
+            String to where to save the volume. If not None that volume will
+            be saved at fname.
+        mri_resolution : bool | tuple
+            Whether to use MRI resolution. If False the morph's resolution
+            will be used. If tuple the voxel size must be given in float values
+            in mm. E.g. mri_resolution=(3., 3., 3.)
+
+        Returns
+        -------
+        img : instance of Nifti1Image
+            The image object.
+        """
+        import nibabel as nib
+        from dipy.align.reslice import reslice
+
+        shape = tuple([int(i) for i in self.morph_data['morph_shape']])
+        affine = self.morph_data['morph_affine']
+        zooms = self.morph_data['morph_zooms'][:3]
+
+        hdr = nib.nifti1.Nifti1Header()
+        hdr.set_xyzt_units('mm', 'msec')
+        hdr['pixdim'][4] = 1e3 * stc.tstep
+
+        if isinstance(mri_resolution, bool):
+            if mri_resolution:
+                new_zooms = self.morph_data['mri_zooms']
+            else:
+                new_zooms = None
```

Comment:
avoid nesting when you can :) Here you can do:

```py
new_zooms = None
if mri_resolution:
   new_zooms = self.morph_data['mri_zooms']
```

Think about the person reading the code. The fewer lines of code and simpler the code, the easier it is for the code to be maintained

Reply:
and what if mri_resolution can also be a tuple (which it can)?
I do now:
```
new_zooms = None

        if isinstance(mri_resolution, bool) and mri_resolution:
            new_zooms = self.morph_data['mri_zooms']

        elif isinstance(mri_resolution, tuple):
            new_zooms = mri_resolution
```